File: cotbeacon/src/main/java/com/jon/beacon/BeaconRepo.java
package com.jon.beacon;

import com.jon.common.AppSpecific;
import com.jon.common.service.CotFactory;
import com.jon.common.service.CotService;
import com.jon.common.ui.ListPresetsActivity;
import com.jon.common.ui.MainFragment;

import java.util.Date;

public class BeaconRepo extends AppSpecific.Repo {

    @Override
    protected MainFragment getMainFragment() {
        return BeaconFragment.getInstance();
    }

    @Override
    protected Date getBuildDate() {
        return BuildConfig.BUILD_TIME;
    }

    @Override
    protected int getBuildVersionCode() {
        return BuildConfig.VERSION_CODE;
    }

    @Override
    protected String getAppId() {
        return BuildConfig.APPLICATION_ID;
    }

    @Override
    protected String getAppName() {
        return context().getString(R.string.app_name);
    }

    @Override
    protected String getPermissionRationale() {
        return context().getString(R.string.permissionRationale);
    }

    @Override
    protected String getVersionName() {
        return BuildConfig.VERSION_NAME;
    }

    @Override
    protected String getPlatform() {
        return context().getString(R.string.appNameAllCaps);
    }

    @Override
    protected boolean isDebug() {
        return BuildConfig.DEBUG;
    }

    @Override
    protected Class<? extends CotService> getCotServiceClass() {
        return BeaconService.class;
    }

    @Override
    protected Class<? extends CotFactory> getCotFactoryClass() {
        return BeaconCotFactory.class;
    }

    @Override
    protected Class<? extends ListPresetsActivity> getListActivityClass() {
        return BeaconListPresetsActivity.class;
    }

    @Override
    public int getSettingsXmlId() {
        return R.xml.settings;
    }

    @Override
    protected int getIconColourId() {
        return R.color.black;
    }
}

File: cotbeacon/src/main/java/com/jon/cotbeacon/BeaconRepo.kt
package com.jon.cotbeacon

import android.content.SharedPreferences
import com.jon.common.CotApplication
import com.jon.common.service.CotService
import com.jon.common.ui.listpresets.ListPresetsActivity
import com.jon.common.ui.main.MainActivity
import com.jon.common.ui.main.MainFragment
import com.jon.common.variants.VariantRepo
import java.util.*

class BeaconRepo : VariantRepo {
    override val mainFragment: MainFragment = BeaconFragment.getInstance()
    override val buildDate: Date = BuildConfig.BUILD_TIME
    override val buildVersionCode = BuildConfig.VERSION_CODE
    override val appId = BuildConfig.APPLICATION_ID
    override val appName = CotApplication.context.getString(R.string.app_name)
    override val permissionRationale = CotApplication.context.getString(R.string.permissionRationale)
    override val versionName = BuildConfig.VERSION_NAME
    override val platform = CotApplication.context.getString(R.string.appNameAllCaps)
    override val isDebug = BuildConfig.DEBUG
    override val cotServiceClass: Class<out CotService> = BeaconService::class.java
    override val mainActivityClass: Class<out MainActivity> = MainActivity::class.java
    override val listActivityClass: Class<out ListPresetsActivity> = BeaconListPresetsActivity::class.java
    override val settingsXmlId = R.xml.settings
    override val iconColourId = R.color.black
    override val menuId = R.menu.beacon_menu

    override fun getCotFactory(prefs: SharedPreferences) = BeaconCotFactory(prefs)
}
